controllers/front/actions.php

```diff
@@ -89,7 +89,7 @@ public function processAdd()
         } elseif (Validate::isEmail((string) Tools::getValue('customer_email'))) {
             $customer_email = (string) Tools::getValue('customer_email');
             $customer = $context->customer->getByEmail($customer_email);
-            $id_customer = (isset($customer->id) && ($customer->id != null)) ? (int) $customer->id : null;
+            $id_customer = (isset($customer->id) && ($customer->id != null)) ? (int) $customer->id : 0;
```

Comment:
@matks It's a legit scenario for notifying users who did not register.

- You come to a product with a watchdog as a public user.
- You are not registered, not logged in, you just enter an email.
- You receive notification when it's back in stock.

I hope it works like this, it does in my watchdog module. :D

Reply:
I agree the behavior is 100% correct. But I am infortable with filling the database with rows using Zero for ID. It's a trick, and sometimes months/weeks later it hits you back. @zuk3975 has encountered such issues I believe.

Reply:
@matks Well, you can have two database tables if you want to sleep better, but IMHO it's not worth it.

If the zero is not a key and there are just no data, why is it a problem that there are zeroes?

The logic is simple...
```php
if (!empty($row['id_customer']) {
  $customer = new Customer($row['id_customer']);
  $email = $customer->email;
} else {
  $email = $row['email'];
}
```

Reply:
@matks I think that is expected in that case.

`id_customer` in `MailAlert` class await an integer in all case.

Anyway, I can change `XX_mailalert_customer_oos` table to make `id_customer` nullable and permit unsigned int with also null value for this purpose. 

As you prefer! But I think it can be prettier to let `id_customer` not null and have a `0` when we don't have find a customer.

Reply:
It is an old module and discussing "clean code" principles in such modules is usually a waste of time :smile:
But In theory I would say it would be better to adapt the `MailAlert::customerHasNotification` contract to accept optional $id_customer if it is actually legit case to handle the customer not found scenario inside it. Because here the NULL value correctly represents the case when customer is not found. 
Or if `MailAlert::customerHasNotification` is not expected to handle the case when customer is not found, then that code should not even be reached when $id_customer is null, and some "customer not found" (or whatever error, or whatever behavior you expect when customer is not found) should be thrown before.

Reply:
Thank you everybody for the answers ;) OK for me then ✅